controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,248 @@
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `yaml:"odfVersion"`
+	OdfDeploymentType   string             `yaml:"odfDeploymentType"`
+	Clients             []ConnectedClient  `yaml:"clients"`
+	StorageCluster      InfoStorageCluster `yaml:"storageCluster"`
+	StorageClusterCount int                `yaml:"storageClusterCount"`
+	StorageSystemName   string             `yaml:"storageSystemName"`
+	IsDROptimized       bool               `yaml:"isDROptimized"`
+}
+
+const (
+	OdfInfoKeyName            = "config.yaml"
+	OdfDeploymentTypeExternal = "external"
+	OdfDeploymentTypeInternal = "internal"
+	RookCephMonSecretName     = "rook-ceph-mon"
+	FsidKey                   = "fsid"
+	OdfOperatorNamePrefix     = "odf-operator"
+	OdfInfoConfigMapName      = "odf-info"
+	OdfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+var mutex sync.RWMutex
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	odfInfoConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OdfInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, odfInfoConfigMap, func() error {
+		if err := ctrl.SetControllerReference(sc, odfInfoConfigMap, r.Scheme); err != nil {
+			return err
+		}
+		r.Log.Info("Creating or updating odf-info configmap", OdfInfoMapKind, klog.KRef(sc.Namespace, OdfInfoConfigMapName))
+		odfInfoData, configErr := getOdfInfoData(r, sc)
+		if configErr != nil {
+			return fmt.Errorf("failed to get ODF info config data: %w", configErr)
+		}
+		odfInfoKey := obj.getOdfInfoKeyName(sc)
+		// Creates or appends to the data map
+		odfInfoConfigMap.Data[odfInfoKey] = odfInfoData
+		return nil
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update odf-info config", OdfInfoMapKind, klog.KRef(sc.Namespace, OdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update odf-info config: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *odfInfoConfig) getOdfInfoKeyName(sc *ocsv1.StorageCluster) string {
+	return sc.Namespace + "/" + sc.Name + "." + OdfInfoKeyName
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	var odfInfoConfigMap corev1.ConfigMap
+	if err = r.Client.Get(r.ctx, types.NamespacedName{Namespace: operatorNamespace}, &odfInfoConfigMap); err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
does this work w/o a name, cm isn't clusterscoped right?

Reply:
Don't we have a `r.get` that read the name and namespace from the passed resources?

Reply:
resolve.